crates/chain/src/tx_graph.rs

```diff
@@ -762,107 +763,190 @@ impl<A: Anchor> TxGraph<A> {
             .map(|r| r.expect("oracle is infallible"))
     }
 
-    /// List outputs that are in `chain` with `chain_tip`.
+    /// Get a filtered list of outputs from the given `outpoints` that are in `chain` with
+    /// `chain_tip`.
     ///
-    /// Floating ouputs are not iterated over.
+    /// `outpoints` is a list of outpoints we are interested in, coupled with the associated txout's
+    /// script pubkey index (`S`).
```

Comment:
why does this have to be a script pubkey index. It's just any data you want to associate with the outpoint. You should suggest if you don't have a natural choice suggest to the reader they can just do `vec_of_outpoints.iter().enumerate()` for example.

The type parameter should probably be named `I` and emphasize that it's there for convenience  and can be `()` if you just don't want one. It would be nice if you could even do without it but it looks like necessary plumbing to make getting the balance easy.

Reply:
How about instead of `I`, we have:

```rust
impl IntoIterator<Item = impl AsRef<OutPoint>>
```

Reply:
Changed generic bound to `OI` (outpoint identifier) and improved docs: ed89de752cf20cb80deefda3ddf65a137b669288